miniProxy.php

```diff
@@ -153,10 +153,20 @@ function($matches) use ($baseURL) {
     $css);
 }
 
+// Parse/extract URL
 $url = substr($_SERVER["REQUEST_URI"], strlen($_SERVER["SCRIPT_NAME"]) + 1);
-if (empty($url)) die("<html><head><title>miniProxy</title></head><body><h1>Welcome to miniProxy!</h1>miniProxy can be directly invoked like this: <a href=\"" . PROXY_PREFIX . "http://google.com/\">" . PROXY_PREFIX . "http://google.com/</a><br /><br />Or, you can simply enter a URL below:<br /><br /><form onsubmit=\"window.location.href='" . PROXY_PREFIX . "' + document.getElementById('site').value; return false;\"><input id=\"site\" type=\"text\" size=\"50\" /><input type=\"submit\" value=\"Proxy It!\" /></form></body></html>");
-if (strpos($url, "//") === 0) $url = "http:" . $url; //Assume that any supplied URLs starting with // are HTTP URLs.
-if (!preg_match("@^.*://@", $url)) $url = "http://" . $url; //Assume that any supplied URLs without a scheme are HTTP URLs.
+if (empty($url)) {
+    die("<html><head><title>miniProxy</title></head><body><h1>Welcome to miniProxy!</h1>miniProxy can be directly invoked like this: <a href=\"" . PROXY_PREFIX . "http://google.com/\">" . PROXY_PREFIX . "http://google.com/</a><br /><br />Or, you can simply enter a URL below:<br /><br /><form onsubmit=\"window.location.href='" . PROXY_PREFIX . "' + document.getElementById('site').value; return false;\"><input id=\"site\" type=\"text\" size=\"50\" /><input type=\"submit\" value=\"Proxy It!\" /></form></body></html>");
+} else if (strpos($url, "//") === 0) {
+    $url = "http:" . $url; //Assume that any supplied URLs starting with // are HTTP URLs.
+} else if (strpos($url, ':/') !== strpos($url, '://')) {
```

Comment:
No, it's not enough to check if the url doesn't contain `://`, it also needs to contain `:/` (otherwise we're talking about a case of a missing schema which is handled by the next else in line number 167)
IIS changes double slash to a single slash, except for the schema at the start of the url. To proxy a url we concatenate it to the miniProxy url, so the schema of the url we're proxying is changed (because it's a part of the original url which already includes a schema). 
For example let's say we entered:
http://example.com/miniProxy.php/http://www.google.com
IIS passes it to the miniProxy like so (notice the missing `/`):
http://example.com/miniProxy.php/http:/www.google.com
So when we look at the url to proxy we have `http:/www.google.com` which contains `:/` but no `://`, the condition captures it and changes it back to `http://www.google.com`.

This issue happened to me on IIS 8.5 and this commit definitely solved it.

Regarding your last question, I guess it's possible, but it seems highly unlikely (having no schema but both `:/` and `://` exists somewhere in the url). It's the same as if the user supplied a url with no schema that contains `://` in it's query string, in which case we won't add `http://` to it and the call will fail.